ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java

```diff
@@ -249,10 +249,15 @@ protected String[] getParamters() {
     return jobEntry.parameters;
   }
 
+  @VisibleForTesting
+  protected JobEntryJob newJobEntryJob() {
+    return new JobEntryJob();
+  }
+
   protected void getParameters( JobMeta inputJobMeta ) {
     try {
       if ( inputJobMeta == null ) {
-        JobEntryJob jej = new JobEntryJob();
+        JobEntryJob jej = newJobEntryJob();
```

Comment:
This is fine and probably a bit easier to understand, but mockito can now mock constructors to help work around issues like this also, just for future reference.

Reply:
Yeah, I tried to solve this through mocking the constructors and didn't work in this case.